src/main/java/io/stargate/sgv2/jsonapi/api/model/command/ValidatableCommandClause.java

```diff
@@ -0,0 +1,128 @@
+package io.stargate.sgv2.jsonapi.api.model.command;
+
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.*;
+
+/**
+ * Interface for any clause of a {@link Command} to implement if it can be validated against a
+ * {@link SchemaObject}.
+ *
+ * <p>Code that wants to validate a clause should call {@link #maybeValidate(CommandContext,
+ * ValidatableCommandClause)} with the clause.
+ *
+ * <p>Example:
+ *
+ * <pre>
+ *  ValidatableCommandClause.maybeValidate(commandContext, command.filterClause());
+ * </pre>
+ */
+public interface ValidatableCommandClause {
+
+  /**
+   * Calls the supplied validatable clause to validate against the {@link SchemaObject} from the
+   * {@link CommandContext} using one of the dedicated validate*Command methods on the interface.
+   *
+   * <p>NOTE: Classes that want to validate a clause should call this method, not the non-static
+   * methods on the interface directly.
+   *
+   * @param commandContext The context the command is running against, including the {@link
+   *     SchemaObject}
+   * @param validatable An object that implements {@link ValidatableCommandClause}, may be null
+   * @param <T> Type of the {@link SchemaObject}
+   */
+  static <T extends SchemaObject> void maybeValidate(
+      CommandContext<T> commandContext, ValidatableCommandClause validatable) {
+    if (validatable == null) {
```

Comment:
Is `null` legal? Assumption being that missing Clause is always valid thing?

Reply:
yes, I think so.